Specs/DateTime/English/DateTimeModel.json

```diff
@@ -21171,6 +21171,118 @@
       }
     ]
   },
+  {
+    "Input": "I want to book a flight for the weekend of halloween",
+    "Context": {
+        "ReferenceDateTime": "2021-11-09T18:00:00"
+    },
+    "NotSupported": "java, javascript, python",
+    "Results": [
+      {
+        "Text": "the weekend of halloween",
+        "Start": 28,
+        "End": 51,
+        "TypeName": "datetimeV2.daterange",
+        "Resolution": {
+          "values": [
+            {
+              "timex": "XXXX-10-31",
+              "type": "daterange",
+              "start": "2021-10-30",
+              "end": "2021-11-01"
+            },
+            {
+              "timex": "XXXX-10-31",
+              "type": "daterange",
+              "start": "2022-10-29",
+              "end": "2022-11-01"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "Input": "I want to book a flight for halloween weekend 2021",
+    "Context": {
+        "ReferenceDateTime": "2021-11-09T18:00:00"
+    },
+    "NotSupported": "java, javascript, python",
+    "Results": [
+      {
+        "Text": "halloween weekend 2021",
+        "Start": 28,
+        "End": 49,
+        "TypeName": "datetimeV2.daterange",
+        "Resolution": {
+          "values": [
+            {
+              "timex": "2021-W43-WE",
+              "type": "daterange",
+              "start": "2021-10-30",
+              "end": "2021-11-01"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "Input": "I need to prepare for labor day weekend",
+    "Context": {
+        "ReferenceDateTime": "2021-11-09T18:00:00"
+    },
+    "NotSupported": "java, javascript, python",
+    "Results": [
+      {
+        "Text": "labor day weekend",
+        "Start": 22,
+        "End": 38,
+        "TypeName": "datetimeV2.daterange",
+        "Resolution": {
+          "values": [
+            {
+              "timex": "XXXX-09-WXX-1-1",
```

Comment:
It seems there's no consistency currently on the returned timexes?

Reply:
This inconsistency stems from the HolidayParser. 

The current behavior is that if the holiday is ambiguous (no year specified), the timex returned by the holiday parser will be used. The holiday parser does not always return the same form of timex, which is the case even if you try the two queries "Halloween" and "Labor day" as individual date inputs. I believe it makes sense to keep the timex format consistent between holiday dates, and holiday weekends. What do you think?